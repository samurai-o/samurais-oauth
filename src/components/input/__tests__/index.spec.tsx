import "jest-styled-components";
import React from "react";
import { render } from "@testing-library/react";
import { Input } from "..";

describe("input", () => {
	it("Input 快照", () => {
		const dom = render(<Input />);
		expect(dom.asFragment()).toMatchSnapshot();
	});

	it("CheckBox 快照", () => {
		const dom = render(<Input placeholder="checkbox" type="checkbox" />);
		expect(dom.asFragment()).toMatchSnapshot();
	});
});