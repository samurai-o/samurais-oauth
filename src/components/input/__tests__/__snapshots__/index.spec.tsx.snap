// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input CheckBox 快照 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  cursor: pointer;
  position: relative;
  width: 18px;
  height: 18px;
  font-size: 12px;
  visibility: hidden;
  overflow: visible;
  color: inherit;
  font-family: inherit;
  line-height: inherit;
  margin: 0;
}

.c1::after {
  position: absolute;
  top: 0px;
  border-radius: 10px;
  border: 2px solid #ddd;
  background-color: transparent;
  color: #000;
  width: 18px;
  height: 18px;
  box-sizing: border-box;
  display: inline-block;
  visibility: visible;
  text-align: center;
  content: ' ';
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
}

.c1:checked::after {
  font-family: system-ui;
  content: '✓';
  font-size: 12px;
  line-height: 1.2;
  text-align: center;
  color: #000;
  font-weight: bold;
  border: 2px solid #000;
  background-color: transparent;
}

<div
    class="c0"
  >
    <input
      class="c1"
      placeholder="checkbox"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;

exports[`input Input 快照 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.1s ease-in 0s;
  transition: all 0.1s ease-in 0s;
  border-radius: 6px;
  border: 2px solid #ddd;
  overflow: hidden;
}

.c0:focus-within {
  border: 2px solid #000;
}

.c1 {
  width: 100%;
  padding: 6px 10px;
  background: transparent;
  border: none;
  font-family: sans-serif;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #252525;
}

.c1:focus {
  outline: none;
  border: none;
}

.c1::-ms-input-placeholder {
  color: #9a9292;
}

.c1::-webkit-input-placeholder {
  color: #9a9292;
}

.c1::-moz-input-placeholder {
  color: #9a9292;
}

<div
    class="c0"
  >
    <input
      class="c1"
    />
  </div>
</DocumentFragment>
`;
